File: src/main/scala/xenit/care4alf/web/JsonObjectArgumentResolver.scala
package xenit.care4alf.web

import org.json.JSONObject
import com.github.dynamicextensionsalfresco.webscripts.arguments.AbstractTypeBasedArgumentResolver
import org.springframework.extensions.webscripts.{WebScriptResponse, WebScriptRequest}
import org.springframework.stereotype.Component

/**
 * Provide JsonObject argument resolution, requires DE 1.0.0
 *
 * @author Laurent Van der Linden
 */
@Component
class JsonObjectArgumentResolver extends AbstractTypeBasedArgumentResolver[JSONObject] {
    override def resolveArgument(request: WebScriptRequest, response: WebScriptResponse): JSONObject = {
        val content = request.getContent().getContent()
        if ("" == content) return null
        try {
            new JSONObject(content)
        }
        catch {
            case exception: Exception => null
        }
    }

    override def getExpectedArgumentType: Class[_] = classOf[JSONObject]
}

File: c4a-impl/src/main/kotlin/eu/xenit/care4alf/web/JSONObjectArgumentResolver.kt
package eu.xenit.care4alf.web

import com.github.dynamicextensionsalfresco.webscripts.arguments.AbstractTypeBasedArgumentResolver
import org.json.JSONObject
import org.springframework.extensions.webscripts.WebScriptRequest
import org.springframework.extensions.webscripts.WebScriptResponse
import org.springframework.stereotype.Component

/**
 * Created by laurentvdl on 11/14/14.
 */
@Component
public class JSONObjectArgumentResolver : AbstractTypeBasedArgumentResolver<JSONObject>() {
    override fun getExpectedArgumentType() = JSONObject::class.java

    override fun resolveArgument(request: WebScriptRequest, response: WebScriptResponse): JSONObject? {
        val content = request.getContent().getContent()
        if ("" == content) return null
        try {
            return JSONObject(content)
        }
        catch(exception: Exception) {
            return null
        }
    }
}